cpp/src/arrow/flight/sql/server.h

```diff
@@ -594,7 +594,16 @@ class ARROW_FLIGHT_SQL_EXPORT FlightSqlServerBase : public FlightServerBase {
   virtual Status EndTransaction(const ServerCallContext& context,
                                 const ActionEndTransactionRequest& request);
 
+  /// \brief Attempt to explicitly cancel a FlightInfo.
+  /// \param[in] context  The call context.
+  /// \param[in] info     The FlightInfo to cancel.
+  /// \return             The cancellation result.
+  virtual arrow::Result<CancelFlightInfoResult> CancelFlightInfo(
+      const ServerCallContext& context, const FlightInfo& info);
```

Comment:
This is not tested.

We may want to add `flight_sql:expiration_time:*` integration tests that use `CancelFlightInfo`/`CloseFlightInfo`/`RefreshFlightEndpoint`. But most of codes will be duplicated with `expiration_time:*` integration tests...

@lidavidm What do you think about this?

Reply:
I don't think we need separate integration tests for Flight SQL if we're testing them in Flight RPC. Just unit tests would be sufficient.

Reply:
OK. Done.